okhttp/src/main/java/okhttp3/OkHttpClient.java

```diff
@@ -844,24 +846,30 @@ public Builder dispatcher(Dispatcher dispatcher) {
      */
     public Builder protocols(List<Protocol> protocols) {
       // Create a private copy of the list.
-      protocols = new ArrayList<>(protocols);
-
-      // Validate that the list has everything we require and nothing we forbid.
-      if (!protocols.contains(Protocol.HTTP_1_1)) {
-        throw new IllegalArgumentException("protocols doesn't contain http/1.1: " + protocols);
+      final Set<Protocol> uniqueProtocols = new HashSet<>(protocols);
```

Comment:
@swankjesse has stated a preference for LinkedHashSet makes these a lot more predictable in tests, e.g. checking the exception message.  But generally you could probably just check the contains(H2C) and size > 1 on the list and fold in duplicates as bad input.  But this isn't super critical code path, so that's minor.

Reply:
Good catch. It wasn't super clear if the list of protocols was treated as a prioritized. Adding to preserve ordering.